src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -232,10 +235,18 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 			if (this.cellModel.currentMode !== CellEditModes.WYSIWYG) {
 				needsTransform = false;
 			} else {
+				let result = undefined;
+				const isFile = URI.parse(linkCalloutResult?.insertUnescapedLinkUrl).scheme === 'file';
+				if (isFile && !path.isAbsolute(linkCalloutResult?.insertUnescapedLinkUrl)) {
+					const notebookDirName = path.dirname(this.cellModel?.notebookModel?.notebookUri.fsPath);
+					const relativePath = (linkCalloutResult?.insertUnescapedLinkUrl).replace(/\\/g, path.posix.sep);
+					result = path.resolve(notebookDirName, relativePath);
+				}
 				// Otherwise, re-focus on the output element, and insert the link directly.
 				this.output?.nativeElement?.focus();
-				// Callout is responsible for returning escaped strings
-				document.execCommand('insertHTML', false, `<a href="${linkCalloutResult?.insertUnescapedLinkUrl}">${linkCalloutResult?.insertUnescapedLinkLabel}</a>`);
+				result = result ?? linkCalloutResult?.insertUnescapedLinkUrl;
+				// Callout is responsible for returning escaped strings.
```

Comment:
This comment doesn't seem to make sense anymore

Reply:
mm but we are scaping the strings, right?

Reply:
That's what I'm saying - this comment implies that the callout is the one doing the escaping. But that's not correct - the caller is the one responsible for doing that